### PR TITLE
Fix the construction of public website urls

### PR DESCRIPTION
**Pull request by jonmorphic**

### CODE REVIEW

**Review by LJBabbage — CHANGES_REQUESTED** [review]
All fine, just would like the tests split up and if we could as team agree to do the Given, When, Then pattern it would be awesome

File: test/support/test_public_website.py [hunk 1]
```diff
@@ -23,39 +23,138 @@ def __getitem__(self, item):
 
 class TestPublicWebsite(TestCase):
 
-    def test_reset_password_url(self):
-
-        config = MockConfig()
+    def test_reset_password_url_excludes_port_80_when_scheme_is_http(self):
```
Comment (andrewmil) [inline test/support/test_public_website.py]: This is really 3 tests in 1 splitting them up would probably make it more readable

File: test/support/test_public_website.py [hunk 2]
```diff
@@ -23,39 +23,138 @@ def __getitem__(self, item):
...
+
+        # When the reset password url is constructed
+        actual_url = unit.reset_password_url('test@email.com')
+
+        # Then it does not contain the port number
+        expected_url_substring = "http://mockhost/passwords/reset-password/"
+        self.assertIn(expected_url_substring, actual_url)
+
+    def test_reset_password_url_excludes_port_443_when_scheme_is_https(self):
+        # Given config contains SCHEME 'http' and PORT 443
```
Comment (andrewmil) [inline test/support/test_public_website.py]: The comments here say `http` instead of `https`

**Review by andrewmil — APPROVED** [review]
Looks like things are working and the testing is comprehensive.

Small comment inline about the structure of the tests but functionally all looks good

File: test/support/test_public_website.py (hunk 2 above)
Comment (jonmorphic) [reply test/support/test_public_website.py]: One of the drawbacks of non-executable Gherkin-like syntax :)

**Review by LJBabbage — APPROVED** [review]

**Pull request merged**